cpp/src/generated/parquet_types.cpp

```diff
@@ -8412,6 +8413,24 @@ std::ostream& operator<<(std::ostream& out, const FileMetaData& obj)
   return out;
 }
 
+// As far as I can tell, the thrift protocol interface does not have a way to directly skip
+// bytes without parsing them.
+// For now, add a hack to access the transport protocol directly to add a
+// skip_bytes method.
+// If we decide to go this way, we can upgrade the thrift library to add this method.
+class CastCompactProtocol {
+public:
+  void *vtbl;
+  ::apache::thrift::transport::TBufferBase *trans_;
+};
```

Comment:
I realize the above is a hack, but this is more for proof of concept. See the discussion below.